src/dmd/dcast.d

```diff
@@ -650,6 +650,16 @@ MATCH implicitConvTo(Expression e, Type t)
                             m = MATCH.convert;
                         result = m;
                         return;
+                    case Tenum:
+                        if ((cast(TypeEnum)tn).sym.isSpecial())
+                        {
+                            /* Allow string literal -> const(wchar_t)[]
+                             */
+                            if (TypeBasic tob = tn.toBasetype().isTypeBasic())
+                            result = tn.implicitConvTo(tob);
+                            return;
```

Comment:
Indentation here looks off. I can't tell whether the early return here should be part of the first or second condition.